.circleci/config.yml

```diff
@@ -0,0 +1,515 @@
+  build-executor:
+    docker:
+      - image: *ci_support_image
+  test-executor:
+    docker:
+      - image: *ci_support_image
+      # start pubsub emulator
+      - image: broadinstitute/study-server-build:pubsub-1
+      - image: circleci/mysql:5.7-ram
+        environment: &DB_VARS
+          MYSQL_ROOT_PASSWORD: rootpw
+          MYSQL_DATABASE: test_db
+          MYSQL_USER: user
+          MYSQL_PASSWORD: passw0rd
+    resource_class: medium+ #default medium not big enough for tests
+
+  deploy-executor:
+    docker: # todo: create new image endpoint, add this in admin tab for faster starts
+    - image: *ci_support_image
+
+commands:
+  compile:
+    description: Run maven compile
+    parameters:
+      m2_repo:
+        type: string
+        default: *m2_repo_path
+      root_path:
+        type: string
+        default: *root_path
+    steps:
+      - restore_cache: *m2_cache_keys
+      - run:
+          name: Run maven
+          command: |
+            # Some images the $HOME directory is not writable. Let's be specific for .m2/repository location
+            mvn compile package -DskipTests -file pom.xml -Dmaven.repo.local=<< parameters.m2_repo >>
+            mvn package -DskipTests -f housekeeping-pom.xml -Dcheckstyle.skip -Dmaven.repo.local=<< parameters.m2_repo >>
+      - save_cache:
+          key: *m2_cache_key
+          paths:
+            - << parameters.root_path >>/.m2
+
+  setup-shared-env:
+    description: "Setup shared ENV vars used by multiple scripts"
+    steps:
+      - run:
+          name: Set environment to be used by build
+          command: |
+            echo "export VAULT_TOKEN=$(vault write -field=token auth/approle/login role_id=$VAULT_ROLE_ID secret_id=$VAULT_ROLE_SECRET_ID)" >> $BASH_ENV
+            echo "export SHORT_GIT_SHA=$(echo '<< pipeline.git.revision >>' | cut -c 1-7)" >> $BASH_ENV
+            source $BASH_ENV
+
+  set-deployment-env:
+    description: "Determine the deploy target and set the ENVIRONMENT var"
+    steps:
+      - run:
+          name: Determine deployment environment for branch << pipeline.git.branch >>
+          command: |
+            case "<< pipeline.git.branch >>" in
+            develop)
+              DEPLOY_ENV=dev
+            ;;
+            # @todo remove when done
+              study-server-parallel-tests)
+              DEPLOY_ENV=dev
+            ;;
+            test)
+              DEPLOY_ENV=test
+            ;;
+            test_deleteme1)
+              DEPLOY_ENV=dev
+            ;;
+            staging)
+              DEPLOY_ENV=staging
+            ;;
+            staging)
+              DEPLOY_ENV=prod
+            ;;
+            *)
+              echo "Cannot map << pipeline.git.branch >> to a deployment environment"
+              exit 17
+            ;;
+            esac
+            echo "Setting deployment ENVIRONMENT to $DEPLOY_ENV"
+            echo "export ENVIRONMENT=$DEPLOY_ENV" >> $BASH_ENV
+            source $BASH_ENV          
+
```

Comment:
Figure out deployment environment based on branch. Store it in environment variable.